content/staging-area.md

```diff
@@ -161,61 +161,134 @@ We give two examples and the instructor can pick one or both:
 
 ## Staging area commands
 
-With the **staging area**, there is one middle point before commits.
+The staging area creates a layer between the working directory and the next
+commit. This adds some complexity but also adds more flexibility to selectively
+prepare commits since **you can modify and stage several times before committing**.
+
+**git add** stages/prepares for the next commit:
+```text
+                git add
+ [project*]  <------------  project*
+                               ^
+                               | modify with editor
+                               |
+  HEAD            .         project
+    |             .
+    |             .
+    |             .
+  HEAD~1          .
+    |             .
+    |             .
+    |             .
+                  .
+  .git            .     working directory
+(history)         .      (files we see)
+```
```

Comment:
I actually quite like this representation!  Somehow it gives me a fresh perspective (if that's good or not is a good question).

Some possibilities to improve it:

Add the column labels on the top.  Separate them from the body with a horizotnal line?
Swap the columns maybe?  So that the logical flow goes left to right.
Hm, is it better for it to be top-to-bottom or bottom-to-top?
Then make a demo that represents `commit -a` - maybe not say that's what it is, but showing project and project*, then the new commit HEAD created from project*.  Basically, a simple example of the syntax before getting into the staging area.

The above ideas are perhaps too much to do for an optional episode (so I'm not saying it should be done), but it is what comes to mind.